src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -488,9 +488,14 @@ public boolean openPairingWindowWithPINCallback(
         deviceControllerPtr, devicePtr, duration, iteration, discriminator, setupPinCode, callback);
   }
 
-  /* Shutdown all cluster attribute subscriptions for a given device */
-  public void shutdownSubscriptions(long devicePtr) {
-    shutdownSubscriptions(deviceControllerPtr, devicePtr);
+  public int getFabricIndex() {
+    return getFabricIndex(deviceControllerPtr);
+  }
+
+  /* Shutdown the subscriptions based upon peerNodeId, fabricIndex, and subscriptionId */
+  public void shutdownSubscriptions(
```

Comment:
Suggestion: For Java, consider splitting this up into separate overloads to make it clear what different combinations are supported. Something like:

```
public void shutdownSubscriptions(int fabricIndex)
public void shutdownSubscriptions(int fabricIndex, long nodeId)
public void shutdownSubscriptions(int fabricIndex, long nodeId, long subscriptionId)
```

It seems like these are the only 3 combinations which are supported?

Reply:
These can then proxy into the nullable native code, but it would be a clearer API for the Java users.

Reply:
You could even omit the first overload and remove the fabricIndex parameter (assuming it's only valid to shutdown subscriptions for your own fabric, that lookup can be done internally to the method).

Reply:
4 combination, another one is shutdownSubscriptions which mean to shutdown all subs. updated, thanks,